app/CPReflectionService/RBObject+CocoaPods.m

```diff
@@ -47,48 +47,56 @@
 }
 
 
-#pragma mark - Errors
+#pragma mark - Conversions
+// Alas, RubyCocoa does not export the ocdata_conv.h header.
+// TODO If we need more polymorphic transforms in the future, see about making that header public.
 
-NSError * _Nonnull
-CPErrorFromException(NSException * _Nonnull exception, NSString * _Nullable message)
+static NSString * _Nullable
+RBString(VALUE rb_string)
 {
-  CPErrorDomainCode code = -1;
+  return rb_string == Qnil ? nil : @(StringValuePtr(rb_string));
+}
 
-  NSMutableDictionary *userInfo = [NSMutableDictionary new];
-  userInfo[CPErrorObjCBacktrace] = exception.callStackSymbols;
-  if (message) {
-    userInfo[NSLocalizedRecoverySuggestionErrorKey] = message;
+static NSArray<NSString *> * _Nonnull
+RBStringArray(VALUE rb_array)
+{
+  long size = RARRAY_LEN(rb_array);
+  NSMutableArray *array = [NSMutableArray arrayWithCapacity:size];
+  for (long i = 0; i < size; i++) {
+    [array addObject:RBString(rb_ary_entry(rb_array, i))];
   }
+  return array;
+}
 
-  if ([exception.name hasPrefix:@"RBException_"]) {
-    VALUE rb_exception = [exception.userInfo[@"$!"] __rbobj__];
 
-    VALUE exceptionNameValue = rb_class_name(rb_obj_class(rb_exception));
-    userInfo[CPErrorName] = @(StringValuePtr(exceptionNameValue));
-    VALUE rb_cause = rb_funcall(rb_exception, rb_intern("cause"), 0);
-    if (rb_cause != Qnil) {
-      exceptionNameValue = rb_class_name(rb_obj_class(rb_cause));
-      userInfo[CPErrorCauseName] = @(StringValuePtr(exceptionNameValue));
-    }
+#pragma mark - Errors
 
-    userInfo[NSLocalizedDescriptionKey] = @"Uncaught Ruby exception.";
-    userInfo[CPErrorRubyBacktrace] = exception.userInfo[CPErrorRubyBacktrace];
+NSError * _Nonnull
+CPErrorFromRubyException(VALUE rb_exception, NSArray * _Nullable objcBacktrace)
+{
+  CPErrorDomainCode code;
+  if (rb_obj_is_kind_of(rb_exception, rb_cPodInformativeError)) {
+    code = CPInformativeError;
+  } else {
+    code = CPStandardError;
+  }
 
-    if (message == nil) {
-      if (rb_obj_is_kind_of(rb_exception, rb_cPodInformativeError)) {
-        code = CPInformativeError;
-      } else {
-        code = CPStandardError;
-      }
-      VALUE messageValue = rb_funcall(rb_exception, rb_intern("message"), 0);
-      messageValue = rb_funcall(messageValue, rb_intern("strip"), 0);
-      userInfo[NSLocalizedRecoverySuggestionErrorKey] = @(StringValuePtr(messageValue));
-    }
+  NSMutableDictionary *userInfo = [NSMutableDictionary new];
+  userInfo[NSLocalizedDescriptionKey] = @"Uncaught Ruby exception.";
```

Comment:
Yeah, the thing is mainly that `NSLocalizedDescriptionKey` by default is used for the heading in an alert and `NSLocalizedRecoverySuggestionErrorKey` for the informative text :/